perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/GroupsManagerBlImpl.java

```diff
@@ -524,7 +524,10 @@ private void logTotallyRemovedMembers(PerunSession sess, Integer parentGroupId,
 
 	@Override
 	public void deleteAllGroups(PerunSession sess, Vo vo) throws GroupAlreadyRemovedException, GroupAlreadyRemovedFromResourceException, GroupRelationDoesNotExist, GroupRelationCannotBeRemoved {
-		for(Group group: getGroupsManagerImpl().getGroups(sess, vo)) {
+		// get parent groups
+		List<Group> groups = getGroupsManagerImpl().getGroups(sess, vo).stream()
+			.filter(group -> !group.getName().contains(":")).collect(Collectors.toList());
```

Comment:
Just a small thing that does not have to be changed:
It would be better to filter according to `getParentGroupId == null` rather than `group.getName().contains(":")`. It would be more readable.

Reply:
Yes, it looks better. I changed it. :+1: